dm/master/openapi.go

```diff
@@ -215,12 +223,70 @@ func (s *Server) DMAPIGetSourceStatus(ctx echo.Context, sourceName string) error
 
 // DMAPIStartTask url is:(POST /api/v1/tasks).
 func (s *Server) DMAPIStartTask(ctx echo.Context) error {
-	return nil
+	var req openapi.CreateTaskRequest
+	if err := ctx.Bind(&req); err != nil {
+		return err
+	}
+	task := req.Task
+	adjustModelTask(&task)
+	// prepare target db config
+	newCtx := ctx.Request().Context()
+	toDBCfg, err := getAndAdjustTaskTargetDBCfg(newCtx, &task)
+	if err != nil {
+		return err
+	}
+	// generate sub task config list
+	subTaskConfigList, err := s.modelToSubTaskConfigList(toDBCfg, &task)
+	if err != nil {
+		return err
+	}
+	// check subtask config
+	subTaskConfigPList := make([]*config.SubTaskConfig, len(subTaskConfigList))
+	for i := range subTaskConfigList {
+		subTaskConfigPList[i] = &subTaskConfigList[i]
+	}
+	if err = checker.CheckSyncConfigFunc(newCtx, subTaskConfigPList,
+		common.DefaultErrorCnt, common.DefaultWarnCnt); err != nil {
+		return terror.WithClass(err, terror.ClassDMMaster)
+	}
+	// end all pre-check, start to create task
+	var (
+		latched = false
+		release scheduler.ReleaseFunc
+	)
+	if req.RemoveMeta {
+		// use same latch for remove-meta and start-task
+		release, err = s.scheduler.AcquireSubtaskLatch(task.Name)
+		if err != nil {
+			return terror.ErrSchedulerLatchInUse.Generate("RemoveMeta", task.Name)
+		}
+		defer release()
+		latched = true
+		err = s.removeMetaData(newCtx, task.Name, *task.MetaSchema, toDBCfg)
+		if err != nil {
+			return terror.Annotate(err, "while removing metadata")
+		}
+	}
+	err = s.scheduler.AddSubTasks(latched, subTaskConfigList...)
+	if err != nil {
+		return err
+	}
+	if release != nil {
+		release()
+	}
+	return ctx.JSON(http.StatusCreated, task)
 }
 
 // DMAPIDeleteTask url is:(DELETE /api/v1/tasks).
 func (s *Server) DMAPIDeleteTask(ctx echo.Context, taskName string) error {
```

Comment:
start and stop task has `-s source` option

Reply:
don't forget this comment

Reply:
already added in https://github.com/pingcap/dm/pull/2141/commits/7c30351eba23b3301b4cc5796bf61c87149bf614